### DISCUSSION TITLE
Any plan to support custom parameters to `init_module()`?

### DISCUSSION THREAD

**Discussion by 0x-j** [aptos-labs/aptos-developer-discussions#273] [Questions] [answered]
### Describe your question in detail.

Currently, if we want to set up some configuration, say the admin address for one that is not the publisher, we need to call an extra function after the module is published. 

If we allow passing in custom parameters to `init_module()` then we don't need the extra function call.

### What have you tried or looked at?  Or how can we reproduce the error?

N/A

### Which operating system are you using?

macOS

### Which SDK or tool are you using? (if any)

Aptos CLI

**Comment by 0x-j** [comment]
Maybe this is a limitation of static dispatch?

**Comment by 0x-j** [comment]
Ok I think we can achieve by setting the creator address in Move.toml and it will be in named address and we access it via `@creator_address` in the contract. 

But that doesn't solve the problem of we want to pass number or other things to init.

**Reply by gregnazario** [reply, marked as answer]
Any other inputs that are passed at creation time for init_module, can be inlined into the code, so it's very rare that there's anything that can't be worked around.

```move
const MY_NUM: u8 = 0;

fun init_module() {
  // Can the devs do something?
  do_something(MY_NUM);
}

fun do_something(num: u8) {
  // ...
}

```

**Reply by 0x-j** [reply]
but that requires us to hardcode a bunch of things just for init
